Updating the page that describes Scripts (#719)
onflow/docs: docs/build/basics/scripts.md

# Scripts

A script provides a light-weight method to query chain data.

> It is an executable Cadence code that can query for Flow execution state data but cannot modify it in any way.

Reviewer: ```suggestion
It is executable Cadence code that can query for Flow execution state data but cannot modify it in any way.
```

# Scripts

A script provides a light-weight method to query chain data.

It is an executable Cadence code that can query for Flow execution state data but cannot modify it in any way.

Unlike a Flow transaction, a script is not signed and requires no transaction fees. Also unlike a transaction, a script can return a value back to the caller.
You can think of executing a script as a read-only operation, very similar to the `eth_call` RPC method on Ethereum.

Scripts are currently executed on either the Access Nodes or the Execution Nodes based on the Access node configuration.

Scripts are defined by the following the Cadence code:

```cadence
// The 'main' function is the entry point function and every script needs to have one.
pub fun main() {
 // Cadence statements to be executed go here
}
```

> Scripts can return a typed value.

Reviewer 2: replace `.` with `:` to align with other segments

```cadence
pub fun main(): Int {
	return 1 + 2
}
```

> Scripts can also accept arguments.

Reviewer 2: replace `.` with `:` to align with other segments

[**Go SDK**](../../tools/clients/flow-go-sdk/index.mdx)

Find a list of all SDKs [here](../../tools/clients/index.md)

### Flow CLI

You can also execute a script by using the [Flow CLI](../../tools/flow-cli/scripts/execute-scripts):

```sh
flow scripts execute ./helloWorld.cdc
```

A user can define their own scripts or can use already defined scripts by the contract authors that can be found by using the [FLIX](../../tools/flow-cli/flix) service.

## Limitations

> 1. **Rate limit** - Script execution is subjected to API rate-limits imposed by the Access nodes and the Execution nodes. The rate limits for the Public Access nodes hosted by QuickNode are mentioned [here](https://www.quicknode.com/docs/flow#endpoint-rate-limits).

Reviewer: ```suggestion
1. **Rate limit** - Script execution is subjected to API rate-limits imposed by the Access nodes and the Execution nodes. The rate limits for the Public Access nodes hosted by QuickNode are outlined [here](https://www.quicknode.com/docs/flow#endpoint-rate-limits).
```

> 2. **Computation limit** - Similar to a transaction, each script is also subjected to a computation limit. Currently, the compute (gas) limit for a script is 100,000.

Reviewer: ```suggestion
2. **Computation limit** - Similar to a transaction, each script is also subjected to a computation limit. The specific value can be configured by individual Access and Execution node operators. Currently, the default compute (gas) limit for a script is 100,000.
```

3. **Historic block data limit**

>    1. Script execution on the execution node is restricted to approximately the last 600 blocks. Any request for script execution on an execution node on a past block (specified by block ID or block height) will fail if that block is more than 600 blocks in the past.

Reviewer: it's actually ~100 blocks these days
```suggestion
   1. Script execution on execution nodes is restricted to approximately the last 100 blocks. Any request for script execution on an execution node on a past block (specified by block ID or block height) will fail if that block is more than 100 blocks in the past.
```

>    2. Script execution on an access node can go much beyond the last 600 blocks but is restricted to the height when the [last](https://developers.flow.com/networks/node-ops/node-operation/past-sporks) network upgrade ([HCU](https://developers.flow.com/networks/node-ops/node-operation/hcu) or spork) occurred. 

Reviewer: ```suggestion
   2. Script execution on an access node can go much beyond the last 100 blocks but is restricted to the height when the [last](https://developers.flow.com/networks/node-ops/node-operation/past-sporks) network upgrade ([HCU](https://developers.flow.com/networks/node-ops/node-operation/hcu) or spork) occurred. 
```
